### PR TITLE
ESQL: Drop unused Node subclasses

### PR DESCRIPTION
**Pull request by nik9000**
And merge NodeSubclassTests into EsqlNodeSubclassTests.

### CODE REVIEW

File: x-pack/plugin/esql-core/src/test/java/org/elasticsearch/xpack/esql/core/tree/NodeSubclassTests.java [hunk 1]
```diff
@@ -6,713 +6,21 @@
...
- * knowledge. As little as I could get away with though.
- * <p>
- * When there are actual tests for a subclass of {@linkplain Node}
- * then this class will do two things:
- * <ul>
- * <li>Skip running any tests for that subclass entirely.
- * <li>Delegate to that test to build nodes of that type when a
- * node of that type is called for.
- * </ul>
+ * Shim to expose protected methods to ESQL proper's NodeSubclassTests.
```
Comment (nik9000) [inline x-pack/plugin/esql-core/src/test/java/org/elasticsearch/xpack/esql/core/tree/NodeSubclassTests.java:14]: This was failing after deleting the unused subclasses. I figured it wasn't worth digging into why and instead merged as much of it as possible into `EsqlNodeSubclassTests`.

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/tree/EsqlNodeSubclassTests.java [hunk 2]
```diff
@@ -58,13 +139,276 @@ public class EsqlNodeSubclassTests<T extends B, B extends Node<B>> extends NodeS
...
+                return b.toString();
+            }
+        } else if (toBuildClass == Like.class) {
+
+            if (argClass == LikePattern.class) {
+                return new LikePattern(randomAlphaOfLength(16), randomFrom('\\', '|', '/', '`'));
+            }
+
+        } else if (argClass == Dissect.Parser.class) {
             // Dissect.Parser is a record / final, cannot be mocked
```
Comment (alex-spies) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/tree/EsqlNodeSubclassTests.java:412]: ++ Nice not to have to jump around between `makeArg` and `pluggableMakeArg` when debugging anymore.

**Review by alex-spies — APPROVED** [review]
LGTM, thanks Nik!

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/tree/EsqlNodeSubclassTests.java [hunk 3]
```diff
@@ -58,13 +139,276 @@ public class EsqlNodeSubclassTests<T extends B, B extends Node<B>> extends NodeS
...
+        /*
+         * Sometimes all of the required type information isn't in the ctor
+         * so we have to hard code it here.
+         */
+        if (toBuildClass == FieldAttribute.class) {
+            // `parent` is nullable.
+            if (argClass == FieldAttribute.class && randomBoolean()) {
+                return null;
+            }
+        } else if (toBuildClass == NodeTests.ChildrenAreAProperty.class) {
```
[the remark this answers is not in the document]
Comment (nik9000) [reply x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/tree/EsqlNodeSubclassTests.java:381]: oh that's... not a phrase I'd have used.

**Pull request merged**